docs: fix naming miss and improve with anchor reference in router guide (#30225)
angular/angular: aio/content/guide/router.md

The `Crisis Detail` route is a child of the `Crisis List`. The router [reuses components](#reuse)
by default, so the `Crisis Detail` component will be re-used as you select different crises.
In contrast, back in the `Hero Detail` route, [the component was recreated](#snapshot-the-no-observable-alternative)

> each time you selected a different hero by visiting the list component previously.

Reviewer: ```suggestion
each time you selected a different hero from the list of heroes.
```
